frappe/email/doctype/notification/notification.js

```diff
@@ -47,28 +48,30 @@ frappe.notification = {
 			frm.set_df_property("date_changed", "options", get_date_change_options());
 
 			let receiver_fields = [];
-			if (frm.doc.channel === "Email") {
-				receiver_fields = $.map(fields, function (d) {
-					// Add User and Email fields from child into select dropdown
-					if (frappe.model.table_fields.includes(d.fieldtype)) {
-						let child_fields = frappe.get_doc("DocType", d.options).fields;
-						return $.map(child_fields, function (df) {
-							return df.options == "Email" ||
-								(df.options == "User" && df.fieldtype == "Link")
-								? get_select_options(df, d.fieldname)
-								: null;
+			let find_receiver_fields = function (extra) {
+				let predicate = function (df) {
+					return extra(df) || (df.options == "User" && df.fieldtype == "Link");
+				};
+				let extract = function (df) {
+					// Add recipients from child doctypes into select dropdown
+					if (frappe.model.table_fields.includes(df.fieldtype)) {
+						let child_fields = frappe.get_doc("DocType", df.options).fields;
+						return $.map(child_fields, function (cdf) {
+							return predicate(cdf) ? get_select_options(cdf, df.fieldname) : null;
 						});
-						// Add User and Email fields from parent into select dropdown
 					} else {
-						return d.options == "Email" ||
-							(d.options == "User" && d.fieldtype == "Link")
-							? get_select_options(d)
-							: null;
+						return predicate(df) ? get_select_options(df) : null;
 					}
+				};
+				return $.map(fields, extract);
```

Comment:
I find this function kinda hard to understand. Let's extract it into the outer scope, add a docstring and rename the parameters to communicate what they do (e.g. `extra` -> `is_receiver_field`)?

Reply:
I made it a little more explicit in 81137aa1e5.

However, I kept in line with the style of the `get_date_change_options` just above and did _not_:
- Extract further into more outer scopes
- Extract the `fields` variable which is accessed from `get_date_change_options` from the outer scope in the same way (which arguably is a bit hard to read, indeed &mdash; but I thought to better stick in line than introduce different coding styles)